disperser/common/inmem/store.go

```diff
@@ -154,6 +154,38 @@ func (q *BlobStore) GetBlobMetadataByStatus(ctx context.Context, status disperse
 	return metas, nil
 }
 
+func (q *BlobStore) GetBlobMetadataByStatusWithPagination(ctx context.Context, status disperser.BlobStatus, limit int32, exclusiveStartKey *disperser.ExclusiveBlobStoreStartKey) ([]*disperser.BlobMetadata, *disperser.ExclusiveBlobStoreStartKey, error) {
+	metas := make([]*disperser.BlobMetadata, 0)
+	foundStart := exclusiveStartKey == nil
+
+	for _, meta := range q.Metadata {
```

Comment:
I don't think the ordering for `q.Metadata` is guaranteed. 
So you may have to sort the keys and iterate it in some order